src/plugins/intel_gpu/src/plugin/ops/reshape.cpp

```diff
@@ -54,7 +54,9 @@ static void CreateCommonReshapeOp(Program& p, const std::shared_ptr<ngraph::Node
     auto reshapePrim = cldnn::reshape(layerName,
                                       reshapeInputId,
                                       outTensor,
-                                      op->get_friendly_name());
+                                      outDims.size(),
```

Comment:
Should we set this in primitive? Cant we just get it from calc_output_layout with outTensor.sizes().size() ?

Reply:
It seems that most of the changes are due to this API change.

Reply:
In case of **bfxyz(3,3,5,4,1)**, the intended format is **format::bfzyx** but, tensor size became **(3,3,5,4,1,1,1,1,1)**. So if we figure out the format this tensor size, the output format became **format::bfyz** and this could lead to "### not match passed format" error.

Reply:
I see that you tried to pass the original rank. By the way as I see the original intention of the reshape seems that if the input/output dims are differnt it should be created as reorder, not reshape. However you are facing that it was created as reshape even though the in/out ranks are differnt, right? Could you elaborate why that happened?

Reply:
For common reshape op, you're right. But, the reshape as a added primitive during other op could be different. The "/src/plugins/intel_gpu/src/plugin/ops/matmul.cpp" file, for example, shows this case. The reshape op added from matmal actually makes some issue on twins model.